packages/router/src/adapters/subgraph/subgraph.ts

```diff
@@ -638,17 +639,30 @@ export const getCancelledFulfilledTransactions = async (routerAddress: string, c
         sendingSideExpiry: Number(cancelledTx.expiry),
       });
 
+      const pre = BigNumber.from(totalLoss.get(cancelledTx.sendingAssetId) ?? 0);
+
+      totalLoss.set(cancelledTx.sendingAssetId, pre.add(cancelledTx.amount));
       return { sendingSide: cancelledTx, receivingSide: receivingSide.transaction };
     }),
   );
 
   miniData.sort((a: any, b: any) => {
     return a.sendingSideExpiry - b.sendingSideExpiry;
   });
+
   const data = {
     records,
     miniData,
   };
   // writeFileSync("./logs.json", JSON.stringify(data));
   console.log(data);
+
+  totalLoss.forEach(async (value, key) => {
```

Comment:
You need to `await` this otherwise totalLoss won't be updated. Do something like 
```
await Promise.all(totalLoss.map(async (value, key) => {...}))
```

Reply:
Also remove the console.log